ci: added precommit hook

uncle-tee opened the pull request:
**Description**

This PR allows us to add a precommit hook to help format based on the prettierc. It will help us standardize the format so that formats will not vary.

bastienbeurier commented:
Thanks a lot, this is great. 

Do you mind giving me a couple details about how this works? I've never used Husky and pre-commit hooks before. Do we need to run anything locally for it to work?

uncle-tee commented:
Husky is an NPM package, which means that we do not need to install anything separately. In the pull request, you will notice that I added a script called "Prepare". This script is essentially a post-script that installs Husky after each time we run NPM install.

However, the downside of this is that Husky will try to install it on production. But as you can see, I have updated the docker file to ignore the "prepare" scripts when running the docker scripts.

**Here's how it works:**

I have added a pre-commit script that is triggered just before a commit is made. It first runs the existing Prettier-format script that we have and then commits the file. With this, we can be sure that the code is formatted before committing. Since we have a Prettier formatter and developers may forget to run the script before pushing, the hook helps to do this automatically

If you have more questions please let me know.

bastienbeurier commented:
Great, thank you for taking the time to explain. This is a really nice addition!
